database/src/procedures/tr_before_survey_sample_period_check_minimum_data_requirements.ts

```diff
@@ -0,0 +1,45 @@
+import { Knex } from 'knex';
+
+/**
+ * Function Name: check_survey_sample_period_minimum_data_requirements
+ *
+ * Trigger Name: tr_before_survey_sample_period_check_minimum_data_requirements
+ *
+ * Affected Tables: survey_sample_period
+ *
+ * Purpose: Asserts that the incoming record meets the minimum data requirements for the survey_sample_period table.
+ * Specifically, checks that at least one of the following columns is not null:
+ * - survey_sample_site_id
+ * - method_technique_id
+ * - start_date, end_date (both must be not null)
+ *
+ * Note: The reason this trigger is necessary is because all of the above columns can be null, by design, but not all of
+ * them at the same time. If all of them were null, then the record would contain no useful information.
+ *
+ * @export
+ * @param {Knex} knex
+ * @return {*}  {Promise<void>}
+ */
+export async function seed(knex: Knex): Promise<void> {
+  await knex.raw(`--sql
+    SET search_path = 'biohub';
+
+    CREATE OR REPLACE FUNCTION biohub.check_survey_sample_period_minimum_data_requirements()
```

Comment:
suggestion: We might be able to replace this trigger with a CHECK contraint using `num_nonnulls`
https://stackoverflow.com/questions/15178859/postgres-constraint-ensuring-one-column-of-many-is-present

Reply:
Will try it out

Reply:
pushed up new constraint (dropped trigger file)